lms/djangoapps/program_enrollments/api/v1/views.py

```diff
@@ -705,3 +716,263 @@ def modify_learner_enrollment_status(self, enrollment_request, program_enrollmen
+
+        user_program_course_enrollments_ids = [enrollment.course_key for enrollment in user_program_course_enrollments]
+        overviews = CourseOverview.get_from_ids_if_exists(user_program_course_enrollments_ids)
+
+        resume_course_run_urls = get_resume_urls_for_enrollments(user, [enrollment.course_enrollment for enrollment in user_program_course_enrollments])
+
+        response = {
+            'course_runs': [],
+        }
+
+        store = modulestore()
+
+        for index, enrollment in enumerate(user_program_course_enrollments):
+            overview = overviews[enrollment.course_key]
+
+            certificate_download_url = None
+            is_certificate_passing = None
+            certificate_info = get_certificate_for_user(user.username, enrollment.course_key)
+
+            if certificate_info:
+                certificate_download_url = certificate_info['download_url']
+                is_certificate_passing = certificate_info['is_passing']
+
+            dates = get_dates_for_course(
+                str(enrollment.course_key),
+                user=user,
+            )
+
+            due_dates = []
+            for (block_key, date_type), date in dates.items():
+                if date_type == 'due':
+                    block = store.get_item(block_key)
+
+                    # get url to the block in the course
+                    block_url = reverse('jump_to', args=[str(enrollment.course_key), block_key])
+                    block_url = request.build_absolute_uri(block_url)
+
+                    due_dates.append({
+                        'name': block.display_name,
+                        'url': block_url,
+                        'date': date,
+                    })
+
+            course_run_dict = {
+                'course_run_id': enrollment.course_key,
+                'display_name': overview.display_name_with_default,
+                'course_status': self.get_course_run_status(overview, is_certificate_passing),
+                'course_run_url': self.get_course_run_url(request, str(enrollment.course_key)),
+                'start_date': overview.start,
+                'end_date': overview.end,
+                'due_dates': due_dates,
+            }
+
+            if certificate_download_url:
+                course_run_dict['certificate_download_url'] = certificate_download_url
+
+            emails_enabled = self.get_emails_enabled(user, str(enrollment.course_key))
+            if emails_enabled:
+                course_run_dict['emails_enabled'] = emails_enabled
+
+            micromasters_title = self.program['title'] if self.program['type'] == 'MicroMasters' else None
+            if micromasters_title:
+                course_run_dict['micromasters_title'] = micromasters_title
+
+            # if the url is '', then the url is None so we can omit it from the response
+            resume_course_run_url = resume_course_run_urls[index] if resume_course_run_urls[index] else None
```

Comment:
isn't this the same as `resume_course_run_url = resume_course_run_urls[index] or None`?

Reply:
No, because I don't want the `resume_course_run_url` key in the dictionary if its value is ''.